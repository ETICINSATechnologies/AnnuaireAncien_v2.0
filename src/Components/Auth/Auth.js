class Auth {
    static connect(jwt){
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(){
        sessionStorage.removeItem('token')
    }

    static isConnected(){
        return !!sessionStorage.getItem('token');
    }

    static getToken() {
        return sessionStorage.getItem('token')
    }

    static getPayload(){
        return sessionStorage.getItem('token');
    }

    static addCorrectButton(buttons){
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
