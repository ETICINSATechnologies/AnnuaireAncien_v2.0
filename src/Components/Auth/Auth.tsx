class Auth {
    static connect(jwt: string) {
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(): void {
        sessionStorage.removeItem('token')
    }

    static isConnected(): boolean {
        return !!sessionStorage.getItem('token');
    }

    static getToken(): string {
        return sessionStorage.getItem('token') as string;
    }

    static addCorrectButton(buttons: Array<string>) {
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
