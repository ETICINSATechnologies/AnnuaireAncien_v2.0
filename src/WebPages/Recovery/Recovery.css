.Recuperation {
    display: grid;

    width: 100%;
    height: 100%;
    text-align: center;
}

.Recuperation .Recuperation_section {
    display: grid;
    grid-template-rows: 15vw 6.7vw 5vw;

    vertical-align: middle;
    justify-self: center;

    margin: auto;

    width: 90%;
    height: 80%;
}

.Recuperation .Recuperation_section * {
    text-align: center;
    justify-self: center;

    margin: 0 auto;
    font-size: 3.7vw;
}

.Recuperation .div2 {
    cursor: pointer;
    color: #009e11;
    background-color: white;
    border: 0;
    text-decoration: underline;
}

.Recuperation .div3 {
    animation-duration: 2s;

    color: white;

    font-size: 2vw;
    margin: 1vw;
    width: 15vw;
}

@keyframes display {
    0% {
        color: white;
    }
    40% {
        color: black;
    }
    100% {
        color: white;
    }
}
