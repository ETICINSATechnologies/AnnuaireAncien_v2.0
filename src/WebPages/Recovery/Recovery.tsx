import React from 'react';
import './Recovery.css';
import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import Auth from "../../Components/Auth/Auth";
import CopyContent from "../../Components/CopyContent/CopyContent";

const Recovery = () => {
    let activeButton = Auth.addCorrectButton(["home"]);

    return (
        <React.Fragment>
            <Header/>
            <Nav buttons={activeButton}> </Nav>
            <div className="Recovery">
                <section className="Recuperation_section">
                    <div className="div1">
                        Afin de récupérer vos identifiants, veuillez contacter par mail l'administrateur de ce site
                        en
                        utilisant
                        l'adresse mail suivante :
                    </div>
                    <CopyContent className="contact" textToCopy="contact@etic-insa.com"/>
                </section>
            </div>
        </React.Fragment>
    );
};

export default Recovery;
