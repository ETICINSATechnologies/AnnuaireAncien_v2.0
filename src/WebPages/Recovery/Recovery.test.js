import React from 'react';

import Recovery from './Recovery';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Auth from '../../Components/Auth/Auth';
import {BrowserRouter} from 'react-router-dom'

Enzyme.configure({adapter: new Adapter()});

describe( 'Recovery component', () => {
    it('renders without crashing', () => {
            const tree = renderer.create(<BrowserRouter><Recovery /></BrowserRouter>).toJSON();
            expect(tree).toMatchSnapshot();
        }
    );

    it('Recovery unit test', () => {
        expect(
            Enzyme.shallow(<Recovery/>)
                .find('div.Recovery')
                .length
        ).toBe(1)
    });

    it('Recovery user connected button test', () => {
            Auth.isConnected = jest.fn(() => true);
            const expected = ["home","disconnection"];
            expect( Auth.addCorrectButton(["home"]) ).toEqual(expected);
        }
    );

    it('Recovery user not connected button test', () => {
            Auth.isConnected = jest.fn(() => false);
            const expected = ["home","connection"];
            expect( Auth.addCorrectButton(["home"]) ).toEqual(expected);
        }
    );

});