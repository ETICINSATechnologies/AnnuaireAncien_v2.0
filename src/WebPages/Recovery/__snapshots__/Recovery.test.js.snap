// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recovery component renders without crashing 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens 
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/"
        onClick={[Function]}
      >
        accueil
      </a>
      <a
        className="button"
        href="/connection"
        onClick={[Function]}
      >
        se connecter
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
      onClick={[Function]}
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <div
    className="Recovery"
  >
    <section
      className="Recuperation_section"
    >
      <div
        className="div1"
      >
        Afin de récupérer vos identifiants, veuillez contacter par mail l'administrateur de ce site en utilisant l'adresse mail suivante :
      </div>
      <div
        className="contact toCopy"
        onClick={[Function]}
        title="Cliquer pour copier "
      >
        contact@etic-insa.com
      </div>
      
    </section>
  </div>,
]
`;
